tests/baselines/reference/completionListForImportAttributes.baseline

```diff
@@ -76,7 +76,7 @@
             },
             {
               "text": "\"resolution-mode\"",
-              "kind": "stringLiteral"
+              "kind": "propertyName"
             },
```

Comment:
This is funky; I'm not sure how much this matters in practice, though...

Reply:
Agreed, but it was already the case that some strings would be marked as things other than text in display parts. 

Before, in symbol printing, some identifiers were created that didn't actually have an identifier valid text (identifier name was a string including delimiters). These identifiers had their symbol set, and during display part emit, they were marked as whatever kind of symbol they had attached. However this did not happene for all string created during symbol printing. So we had some parts of symbol names as identifiers with symbol and invalid text, and some as string literals with no symbol. 


Now, I create string literals and number literals for all symbol names which are not valid identifiers and set the symbol on those too. This does mean that some parts of the symbol name now have the symbol attached to then (where before they did not) and can have the correct kind set in display parts.

These are the changes in `symbolToExpression` in checker.ts and the changes to `emitLiteral` in emitter.ts